src/main/java/org/broadinstitute/hellbender/utils/haplotype/HaplotypeBAMDestination.java

```diff
@@ -45,14 +61,26 @@ protected HaplotypeBAMDestination(SAMFileHeader sourceHeader, final String haplo
         bamOutputHeader.setReadGroups(readGroups);
 
         bamOutputHeader.addProgramRecord(new SAMProgramRecord("HalpotypeBAMWriter"));
+
+        samWriter = new SAMFileGATKReadWriter(ReadUtils.createCommonSAMWriter(
+                outPath,
+                null,
+                getBAMOutputHeader(), // use the header derived from the source header by HaplotypeBAMDestination
+                false,
+                createBamOutIndex,
+                createBamOutMD5
+        ));
     }
 
     /**
      * Write a read to the output specified by this destination.
      *
      * @param read the read to write out
      */
-    public abstract void add(final GATKRead read);
+    public void add(final GATKRead read){
+        Utils.nonNull(read, "read cannot be null");
+        samWriter.addRead(read);
+    };
```

Comment:
There are some extraneous semicolons at the ends of method bodies (here and below). These are not needed now that the methods are no longer abstract.